homeassistant/components/fujitsu_hvac/climate.py

```diff
@@ -0,0 +1,205 @@
+
+from homeassistant.components.climate import (
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACMode,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_TEMPERATURE, PRECISION_HALVES, UnitOfTemperature
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import (
+    API,
+    DOMAIN,
+    FUJI_TO_HA_FAN,
+    FUJI_TO_HA_HVAC,
+    FUJI_TO_HA_SWING,
+    HA_TO_FUJI_FAN,
+    HA_TO_FUJI_HVAC,
+    HA_TO_FUJI_SWING,
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up one Fujitsu HVAC device."""
+    api = hass.data[DOMAIN][API]
+
+    devices = await api.async_get_devices()
+    devices = list(filter(lambda x: isinstance(x, FujitsuHVAC), devices))
+    await gather(*[dev.async_update() for dev in devices])
+
+    async_add_entities([FujitsuHVACDevice(dev) for dev in devices])
+
+
+class FujitsuHVACDevice(ClimateEntity):
+    """Represent a Fujitsu HVAC device."""
+
+    _attr_temperature_unit = UnitOfTemperature.CELSIUS
+    _attr_precision = PRECISION_HALVES
+    _attr_target_temperature_step = 0.5
+    _attr_has_entity_name = True
+    _attr_name = None
+
+    def __init__(self, dev: FujitsuHVAC) -> None:
+        """Set up static attributes."""
+        self._dev = dev
+
+        self._attr_unique_id = dev.device_serial_number
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, dev.device_serial_number)},
+            name=dev.device_name,
+            manufacturer="Fujitsu",
+            model=dev.property_values["model_name"],
+            serial_number=dev.device_serial_number,
+            sw_version=self._dev.property_values["mcu_firmware_version"],
+        )
+
+    async def async_update(self) -> None:
+        """Gather data from API and update our attributes."""
+        try:
+            if self._dev.ayla_api.token_expiring_soon:
+                await self._dev.ayla_api.async_refresh_auth()
+        except AylaAuthError:
+            await self._dev.ayla_api.async_sign_in()
+
+        await self._dev.async_update()
+
+    @property
+    def fan_mode(self) -> str | None:
+        """Return the fan setting.
+
+        Requires ClimateEntityFeature.FAN_MODE.
+        """
+        with suppress(KeyError):
+            return FUJI_TO_HA_FAN[self._dev.fan_speed]
+
+        return None
+
+    @property
+    def fan_modes(self) -> list[str] | None:
+        """Return the list of available fan modes.
+
+        Requires ClimateEntityFeature.FAN_MODE.
+        """
+        ret = [
+            FUJI_TO_HA_FAN[mode]
+            for mode in self._dev.supported_fan_speeds
+            if mode in FUJI_TO_HA_FAN
+        ]
+
+        if len(ret) > 0:
+            return ret
+
+        return None
```

Comment:
Euh, I am not 100% sure on this, what's the difference between returning an empty list and None?